### PR TITLE
[Merged by Bors] - feat: port Data.Dfinsupp.Order

### PR DESCRIPTION
**Pull request by int-y1**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
Remaining tasks:

* ``unused variable `i` [linter.unusedVariables]`` for lines like `variable (α) [∀ i, Zero (α i)]`. This looks like a bug in the linter.
* Fix `disjoint_iff`.

**Comment by nomeata** [comment]
Confirmed, and looks like a bug to me. Likely https://github.com/leanprover/lean4/issues/2088

I suggest to disable the linter for now, and merge this to make progress, and enable again once https://github.com/leanprover/lean4/issues/2088 is fixed.

File: Mathlib/Data/Dfinsupp/Order.lean [hunk 1]
```diff
@@ -17,13 +17,19 @@ This file lifts order structures on the `α i` to `Π₀ i, α i`.
...
 
+-- porting note: The unusedVariables linter has false positives in code like
+--    variable (α) [∀ i, Zero (α i)]
+-- marking the  `i` as unused. So disable that lint for now.
+-- Likely issue https://github.com/leanprover/lean4/issues/2088, so remove when that is fixed
+set_option linter.unusedVariables false
 
-open BigOperators
+-- Porting note: Unkonwn namespace BigOperators
+-- open BigOperators
```
Comment (jcommelin) [inline Mathlib/Data/Dfinsupp/Order.lean:32]: Please merge master and re-enable this locale/namespace.

File: Mathlib/Data/Dfinsupp/Order.lean [hunk 2]
```diff
@@ -17,13 +17,19 @@ This file lifts order structures on the `α i` to `Π₀ i, α i`.
...
+* `Dfinsupp.orderEmbeddingToFun`: The order embedding from finitely supported dependent functions
   to functions.
 
 -/
 
+-- porting note: The unusedVariables linter has false positives in code like
+--    variable (α) [∀ i, Zero (α i)]
+-- marking the  `i` as unused. So disable that lint for now.
+-- Likely issue https://github.com/leanprover/lean4/issues/2088, so remove when that is fixed
+set_option linter.unusedVariables false
```
Comment (jcommelin) [inline Mathlib/Data/Dfinsupp/Order.lean:29]: Please use the syntax
```
set_option linter.unusedVariables false in
theorem/def fooBar (α) [∀ i, Zero (α i)] ....
```
to disable the linter locally.

Comment (nomeata) [reply Mathlib/Data/Dfinsupp/Order.lean:29]: Applying this to `variables` lines seems to change their semantics, and the later code turn red. 

I would not waste time on this, not linting for unused variables on the file doesn’t hurt, and we’ll remove it once the bug is fixed anways.

File: Mathlib/Data/Dfinsupp/Order.lean (hunk 1 above)
Comment (nomeata) [reply Mathlib/Data/Dfinsupp/Order.lean:32]: Even if it is not used? 
Or did you change things in the file because `BigOperators` was not there, @int-y1 ?

Comment (int-y1) [reply Mathlib/Data/Dfinsupp/Order.lean]: Re-enabled. I didn't notice anything related to `BigOperators` while fixing the file.

> Even if it is not used?

For now, it's probably fine to keep `BigOperators`.

File: Mathlib/Data/Dfinsupp/Order.lean (hunk 2 above)
Comment (int-y1) [reply Mathlib/Data/Dfinsupp/Order.lean]: Apparently, the linter is happy with:
```
variable (α)
variable [∀ i, Zero (α i)]
```

So I did that, then deleted `set_option`.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**